packages/plugin-ext/src/hosted/browser/hosted-plugin-watcher.ts

```diff
@@ -27,10 +29,25 @@ export class HostedPluginWatcher {
     private readonly onDidDeployEmitter = new Emitter<void>();
     readonly onDidDeploy = this.onDidDeployEmitter.event;
 
+    readonly clientId = UUID.uuid4();
+
+    protected _initialized = new Deferred<void>();
+    get initialized(): Promise<void> {
+        return this._initialized.promise;
+    }
+
+    reset(): void {
+        this._initialized = new Deferred<void>();
```

Comment:
Should we also generate a new id here, just to avoid confusion with previous connections?

Reply:
I think it will be confusing if the same window will get different id on reconnect, e.g. you see that some plugin was disconnected for some client id, but then it gets immediately started for another client id without first loading it. I think it would be more consistent to see that a plugin was disconnected for such client and then started again for the same client.